### PR TITLE
Juju 7288/add prom metric for jimm methods

### PR DESCRIPTION
**Pull request by SimoneDutto**
## Description

Add Prometheus for jimm's methods.
In particular in this pr, I've added the tracking for Cleanup functions.

> fly-by update name of one jimm's method (second commit)

Fixes _JIRA/GitHub issue number_

## Engineering checklist
*Check only items that apply*

- [ ] Documentation updated
- [ ] Covered by unit tests
- [ ] Covered by integration tests

### CODE REVIEW

**Review by ale8k — APPROVED** [review]

File: internal/jimm/access.go [hunk 1]
```diff
@@ -723,10 +724,14 @@ func (j *JIMM) parseAndValidateTag(ctx context.Context, key string) (*ofganames.
 //
 // This approach to cleaning up tuples is intended to be temporary while we implement
 // a better approach to eventual consistency of JIMM's database objects and OpenFGA tuples.
-func (j *JIMM) OpenFGACleanup(ctx context.Context) error {
+func (j *JIMM) OpenFGACleanup(ctx context.Context) (err error) {
+	const op = errors.Op("jimm.CleanupDyingModels")
+	zapctx.Info(ctx, string(op))
+	durationObserver := servermon.DurationObserver(servermon.JimmMethodsDurationHistogram, string(op))
+	defer durationObserver()
+	defer servermon.ErrorCounter(servermon.DBQueryErrorCount, &err, string(op))
```
Comment (alesstimec) [inline internal/jimm/access.go]: why are we using the `DBQueryErrorCount` here? i don't think this is what the metric was intended to measure.

File: internal/jimm/model_cleanup.go [hunk 2]
```diff
@@ -12,15 +12,19 @@ import (
...
 
 // CleanupDyingModels loops over dying models, contacting the respective controller.
 // And deleting the model from our database if the error is `NotFound` which means the model was successfully deleted.
-func (j *JIMM) CleanupDyingModels(ctx context.Context) error {
+func (j *JIMM) CleanupDyingModels(ctx context.Context) (err error) {
 	const op = errors.Op("jimm.CleanupDyingModels")
 	zapctx.Info(ctx, string(op))
+	durationObserver := servermon.DurationObserver(servermon.JimmMethodsDurationHistogram, string(op))
+	defer durationObserver()
+	defer servermon.ErrorCounter(servermon.DBQueryErrorCount, &err, string(op))
```
Comment (alesstimec) [inline internal/jimm/model_cleanup.go]: again, this is not the right metric to use here. `DBQueryErroCount` is intended for database query errors only.

**Review by alesstimec — CHANGES_REQUESTED** [review]

File: internal/jimm/access.go (hunk 1 above)
Comment (SimoneDutto) [reply internal/jimm/access.go]: you are right. removed it

File: internal/jimm/model_cleanup.go (hunk 2 above)
Comment (SimoneDutto) [reply internal/jimm/model_cleanup.go]: you are right. removed it

**Review by alesstimec — APPROVED** [review]

**Pull request merged**